masonry/src/text/selection.rs

```diff
@@ -206,16 +188,13 @@ impl<T: Selectable> TextWithSelection<T> {
 
     /// Call when this widget becomes focused
     pub fn focus_gained(&mut self) {
-        if self.selection.is_none() {
-            // TODO - We need to have some "memory" of the text selected instead.
-            self.selection = Some(Selection::caret(self.text().len(), Affinity::Downstream));
-        }
+        self.selection_visible = true;
         self.needs_selection_update = true;
     }
 
     /// Call when another widget becomes focused
     pub fn focus_lost(&mut self) {
-        self.selection = None;
+        self.selection_visible = false;
```

Comment:
Is there ever a time that this is different from the parent widget being focused? I see that it is used in both `rebuild_with_attributes` and `draw`, so passing it in from the parent would be messy.